examples/coupled_quartic_hamiltonian.py

```diff
@@ -0,0 +1,656 @@
+                      events = lambda t,x: half_period_model(t, x, par), rtol=RelTol, atol=AbsTol)
+    te2 = soln2.t_events[0]
+    t2 = [0,te2[n_turn]]#[0,te2[1]]
+    turn2 = soln2.sol(t2)
+    x_turn2 = turn2[0,-1] 
+    y_turn2 = turn2[1,-1] 
+    x_diff2 = guess2[0] - x_turn2
+    y_diff2 = guess2[1] - y_turn2
+    
+
+    print("Initial guess1%s, initial guess2 %s, x_diff1 is %s, x_diff2 is %s " %(guess1, \
+                                                                                 guess2, x_diff1, \
+                                                                                 x_diff2))
+    
+    return x_diff1, x_diff2 #,y_diff1, y_diff2
+
+
+def guess_coords_coupled(guess1, guess2, i, n, e, get_coord_model, par):
+    """
+    Returns x and y (configuration space) coordinates as guess for the next iteration of the 
+    turning point based on confifuration difference method
+    """
+    
+    
+    h = (guess2[0] - guess1[0])*i/n
+    print("h is ",h)
+    xguess = guess1[0]+h
+    f = lambda y: get_coord_model(xguess,y,e,par)
+    yanalytic = math.sqrt(2/(par[1]+par[6]))*(-math.sqrt( e +0.5*par[3]* xguess**2 - \
+                         0.25*par[4]*xguess**4 -0.5*par[6]* xguess**2 + \
+                         (par[6]*xguess)**2/(2*(par[1] +par[6]) )) + 
+                        par[6]/(math.sqrt(2*(par[1]+par[6])) )*xguess ) #coupled
+    yguess = optimize.newton(f,yanalytic) 
+    
+    return xguess, yguess
+
+
+def plot_iter_orbit_coupled(x, ax, e, par):
+    """ 
+    Plots the orbit in the 3D space of (x,y,p_y) coordinates with the initial and 
+    final points marked 
+    """
+    
+#    label_fs = 10
+    axis_fs = 15 # fontsize for publications 
+    
+    ax.plot(x[:,0],x[:,1],x[:,3],'-')
+    ax.plot(x[:,0],x[:,1],-x[:,3],'--')
+    ax.scatter(x[0,0],x[0,1],x[0,3],s=20,marker='*')
+    ax.scatter(x[-1,0],x[-1,1],x[-1,3],s=20,marker='o')
+    ax.set_xlabel(r'$x$', fontsize=axis_fs)
+    ax.set_ylabel(r'$y$', fontsize=axis_fs)
+    ax.set_zlabel(r'$p_y$', fontsize=axis_fs)
+    ax.set_title(r'$\Delta E$ = %e' %(np.mean(e) - par[2]) ,fontsize=axis_fs)
+    #par(3) is the energy of the saddle
+    ax.set_xlim(-0.1, 0.1)
+    
+    return 
+
+
+def ham2dof_coupled(t, x, par):
+    """ Returns the Hamiltonian vector field (Hamilton's equations of motion) """
+    
+    xDot = np.zeros(4)
+    
+    dVdx = (-par[3]+par[6])*x[0]+par[4]*(x[0])**3-par[6]*x[1]
+    dVdy = (par[5]+par[6])*x[1]-par[6]*x[0]
+        
+    xDot[0] = x[2]/par[0]
+    xDot[1] = x[3]/par[1]
+    xDot[2] = -dVdx 
+    xDot[3] = -dVdy
+    
+    return list(xDot)    
+
+def half_period_coupled(t, x, par):
+    """ 
+    Returns the coordinate for the half-period event for the unstable periodic orbit                          
+    
+    xDot = x[0]
+    yDot = x[1]
+    pxDot = x[2]
+    pyDot = x[3]
+    """
+    
+    terminal = True
+    # The zero can be approached from either direction
+    direction = 0 #0: all directions of crossing

```

Comment:
What are `terminal` and `direction` used for?

Reply:
It's used for event integration. We have updated this feature.